### PR TITLE
bitwarden-desktop: add aarch64-linux support 

### PR DESCRIPTION
**Pull request by jack-wines**
I added aarch64-linux for bitwarden's desktop client. I've tested it and it works without issue. The only changes required relate to two directory names that are platform dependent.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
LGTM

Please make sure your commit message follows CONTRIBUTING.md

File: pkgs/by-name/bi/bitwarden-desktop/package.nix [hunk 1]
```diff
@@ -93,6 +94,11 @@ in buildNpmPackage rec {
     libsecret
   ];
 
+  bitwardenDesktopNativeArch = {
+    aarch64 = "arm64";
+    x86_64  = "x64";
+  }.${stdenv.hostPlatform.parsed.cpu.name};
```
Comment (mweinelt) [inline pkgs/by-name/bi/bitwarden-desktop/package.nix:100]:
```suggestion
  bitwardenDesktopNativeArch = {
    aarch64 = "arm64";
    x86_64  = "x64";
  }.${stdenv.hostPlatform.parsed.cpu.name} or throw "bitwarden-desktop: unsupported CPU name ${stdenv.hostPlatform.parsed.cpu.name}";;
```

Make sure we give proper error messages for everyone else.

**Comment by jackoe** [comment]
hold up going to squash those two commits together

**Comment by jackoe** [comment]
squashed

**Review by amarshall — COMMENTED** [review]
Thanks! I don’t have aarch64 platform available so cannot easily test.

Please also update the PR title to match the commit message format, and then one other small comment.

File: pkgs/by-name/bi/bitwarden-desktop/package.nix [hunk 2]
```diff
@@ -1,4 +1,5 @@
-{ lib
+{ stdenv
```
[the remark this answers is not in the document]
Comment (jackoe) [reply pkgs/by-name/bi/bitwarden-desktop/package.nix]: done :)

**Review by mweinelt — APPROVED** [review]

**Review by amarshall — APPROVED** [review]
Code looks good. One thing I’ll note is that upstream doesn’t, afaik, officially support aarch64-linux (and I don’t have a dev env for it either), so support for the platform here will likely be limited.

File: pkgs/by-name/bi/bitwarden-desktop/package.nix [hunk 3]
```diff
@@ -93,6 +94,11 @@ in buildNpmPackage rec {
     libsecret
   ];
 
+  bitwardenDesktopNativeArch = {
+    aarch64 = "arm64";
+    x86_64  = "x64";
+  }.${stdenv.hostPlatform.parsed.cpu.name} or throw "bitwarden-desktop: unsupported CPU family ${stdenv.hostPlatform.parsed.cpu.name}";
```
Comment (mweinelt) [inline pkgs/by-name/bi/bitwarden-desktop/package.nix:100]:
```suggestion
  bitwardenDesktopNativeArch = {
    aarch64 = "arm64";
    x86_64  = "x64";
  }.${stdenv.hostPlatform.parsed.cpu.name} or (throw "bitwarden-desktop: unsupported CPU family ${stdenv.hostPlatform.parsed.cpu.name}");
```

and move this up into the let block, please.

**Review by mweinelt — CHANGES_REQUESTED** [review]

File: pkgs/by-name/bi/bitwarden-desktop/package.nix (hunk 3 above)
Comment (jackoe) [reply pkgs/by-name/bi/bitwarden-desktop/package.nix]: done :+1:

**Review by mweinelt — APPROVED** [review]

**Pull request merged**